### PR TITLE
Jvenstad/contain zms zts et al

### PR DESCRIPTION
**Pull request by jonmv**
@tokle FYI.

### CODE REVIEW

**Comment by jonmv** [comment]
@tokle tests pass now. Take a look if you want. Otherwise, I can try to split into smaller commit first, but it won't be easy >_<

**Comment by jonmv** [comment]
We also need to set up the new component in controller services.

**Comment by jonmv** [comment]
Moved things to facilitate component setup, and added some more scaffolding for you to fill out.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/TenantController.java [hunk 1]
```diff
@@ -42,36 +40,19 @@
 
     private final Controller controller;
     private final CuratorDb curator;
-    private final ZmsClientFacade zmsClient;
-    private final ZtsClient ztsClient;
-    private final AthenzService controllerIdentity;
+    private final PermitStore permits;
```
Comment (tokle) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/TenantController.java]: What we really create here are roles. We should consider a different name

AccessControlBackend ?

**Review by tokle — CHANGES_REQUESTED** [review]

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerControllerTester.java [hunk 2]
```diff
@@ -132,7 +142,7 @@ private AthenzDomain addTenantAthenzDomain(String domainName, String userName) {
         AthenzDomain athensDomain = new AthenzDomain(domainName);
         AthenzDbMock.Domain domain = new AthenzDbMock.Domain(athensDomain);
         domain.markAsVespaTenant();
-        domain.admin(AthenzIdentities.from(new AthenzDomain("domain"), userName));
+        domain.admin(AthenzIdentities.from(new AthenzDomain("user"), userName));
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerControllerTester.java]: Done.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 3]
```diff
@@ -217,7 +217,7 @@ public void setGlobalRotationStatus(DeploymentId deployment, EndpointStatus stat
      *
      * @throws IllegalArgumentException if the application already exists
      */
-    public Application createApplication(ApplicationId id, Optional<OktaAccessToken> token) {
+    public Application createApplication(ApplicationId id, Optional<ApplicationPermit> permit) {
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Ah, but the application thing also contains the principal creating the application, which shall be the first admin for the `Cloud`s, while the `Athenz` one doesn't have this. So we're on `Spec` again, I think.

**Comment by jonmv** [comment]
Do you mind if I merge this now to see if it works, and then rename things later?

**Review by tokle — APPROVED** [review]

**Pull request merged**